File: frontend-v2/src/api/insight.ts
import { apiGet } from "./client";

/**
 * "Hızlı Özet" panelinin dört kartı.
 *
 * `backend/app/schemas/insight.py` ile birebir eşleşir. Kartları üreten
 * Özet Ajanı'nın sözleşmesi için `agents/summary_agent.py` docstring'i:
 * ajan kendi sayısını üretemez, ürettiği metindeki her sayı deterministik
 * bloğa karşı doğrulanır.
 */

/** Akordeon sırası bu id'lere göre kurulur. */
export type InsightCardId = "genel" | "portfoy" | "piyasa" | "risk";

export interface ApiInsightCard {
  id: InsightCardId;
  title: string;
  body: string;
  /**
   * Kartın metni LLM'den DEĞİL, deterministik özetten geliyor: ya sayı
   * doğrulaması başarısız oldu ya da veri kaynağı düştü. Kart yine
   * gösterilir (zarif düşüş) ama arayüz bunu belirtmelidir — cilalı bir
   * cümle beklerken ham satırlar gören kullanıcı, sebebini bilmeli.
   */
  degraded: boolean;
}

export interface ApiInsightCards {
  user_id: string;
  generated_at: string;
  cards: ApiInsightCard[];
}

/**
 * İSTEMCİ ZAMAN AŞIMI 70 SANİYE — `apiGet`in 8 sn'lik varsayılanı BU UÇ İÇİN
 * YANLIŞTI ve "özet alınamadı" hatasının gerçek sebebiydi (ölçüldü, 2 Eylül
 * 2026).
 *
 * Diğer uçlar tek bir DB sorgusu; 8 sn onlar için makul bir üst sınır. Bu uç
 * ise beş MCP tool'u + bir LLM turu çalıştırıyor ve 10-30 saniye sürüyor.
 * Tarayıcı 8. saniyede isteği iptal edip hata gösteriyordu — backend'in
 * cevabı yolda olsa bile.
 *
 * Sunucu tarafındaki sınır 60 sn (`backend/app/api/insight.py`). Buradaki
 * değer ondan BÜYÜK olmalı: aksi hâlde istemci önce vazgeçer ve kullanıcı,
 * sunucunun dürüst "üretilemedi" yanıtı yerine genel bir bağlantı hatası
 * görür. 70 = 60 + ağ payı.
 */
const ZAMAN_ASIMI_MS = 70_000;

/**
 * Panel HER AÇILIŞTA yeniden üretilir; önbellek YOK (ürün kararı, 1 Eylül
 * 2026: paneli açmak yenilemek demek). Bu yüzden burada da, hook'ta da
 * saklama katmanı bulunmuyor — diğer ekranlarda olan modül düzeyi önbellek
 * kalıbı bilerek uygulanmadı.
 */
export function fetchInsightCards(userId: string): Promise<ApiInsightCards> {
  return apiGet<ApiInsightCards>(`/api/insight/${userId}`, { timeoutMs: ZAMAN_ASIMI_MS });
}

File: backend/app/schemas/insight.py
"""Hızlı Özet paneli — dört kart.

Kartların üretimi `agents/summary_agent.py`'de; buradaki şema yalnızca uç
sözleşmesi. Ajanın "kendi sayısı olmayan ajan" tanımı ve `degraded` alanının
anlamı için o modülün docstring'ine bakın.
"""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class InsightCard(BaseModel):
    model_config = ConfigDict(frozen=True)

    # `genel | portfoy | piyasa | risk` — arayüz akordeon sırasını buna göre
    # kuruyor ve açılışta bulunulan sayfanın kartını genişletiyor.
    id: str
    title: str
    body: str
    # LLM metni sayı doğrulamasından geçemedi ya da tool'u düştü; gövde
    # deterministik özet. KART DÜŞMÜYOR — kullanıcı boş panel görmemeli
    # (zarif düşüş) — ama arayüz bunu belirtebilsin diye işaretleniyor.
    degraded: bool = False


class InsightCards(BaseModel):
    model_config = ConfigDict(frozen=True)

    user_id: str
    # Panel HER AÇILIŞTA yeniden üretiliyor (ürün kararı, 1 Eylül 2026):
    # oturum önbelleği yok, dolayısıyla bu damga "bu metin ne zaman yazıldı"
    # sorusunun cevabıdır. Verinin kendi tarihi kartların içinde yazılı.
    generated_at: datetime
    cards: list[InsightCard]
